packages/core/src/index.ts

```diff
@@ -11,7 +11,11 @@ export {
 } from './error/TRPCError';
 
 export { initTRPC } from './initTRPC';
-export type { MiddlewareFunction, MiddlewareBuilder } from './middleware';
+export type {
+  AnyMiddlewareFunction,
+  MiddlewareFunction,
+  MiddlewareBuilder,
+} from './middleware';
```

Comment:
Need at least AnyMiddlewareFunction in order to continue rolling the Middleware Appliers pattern in userland. Though export just the `Next` function type as in #5112 would be the productionised approach.